File: DotNet/CesiumLanguageWriter/Advanced/CesiumPathModeValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumPathModeValuePropertyWriter"/> to implement
    /// <see cref="ICesiumInterpolatableValuePropertyWriter{TValue}"/> for <see cref="CesiumPathMode"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumPathModeValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumPathModeValuePropertyAdaptor<TFrom> : CesiumWriterAdaptor<TFrom, CesiumPathMode>
        where TFrom : class, ICesiumPathModeValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="CesiumPathMode"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumPathModeValuePropertyAdaptor([NotNull] TFrom parent,
                                                  [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, CesiumPathMode> writeValueCallback,
                                                  [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeDeleteValueCallback)
        {
        }
    }
}

File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumPathModeValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 * Adapts a class that implements {@link ICesiumPathModeValuePropertyWriter} to implement
 {@link ICesiumInterpolatableValuePropertyWriter} for {@link CesiumPathMode} values.
 * @param <TFrom> The class that implements {@link ICesiumPathModeValuePropertyWriter} to adapt.
 */
@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
public class CesiumPathModeValuePropertyAdaptor<TFrom extends ICesiumPathModeValuePropertyWriter> extends CesiumWriterAdaptor<TFrom, CesiumPathMode> {
    /**
    * Initializes a new instance.
    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link CesiumPathMode}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumPathModeValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, CesiumPathMode> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeDeleteValueCallback);
    }
}